jsonrpc/server.go

```diff
@@ -344,3 +334,24 @@ func buildArguments(params, handler any, configuredParams []Parameter) ([]reflec
 	}
 	return args, nil
 }
+
+func parseParam(param any, t reflect.Type) (reflect.Value, error) {
+	handlerParam := reflect.New(t)
+	valueMarshaled, err := json.Marshal(param) // we have to marshal the value into JSON again
```

Comment:
why do we need to marshal again?

Reply:
JSON RPC allows parameters to be sent as a list or an object.

```
{"jsonrpc": "2.0", "method": "subtract", "params": [42, 23], "id": 1}
{"jsonrpc": "2.0", "method": "subtract", "params": {"subtrahend": 23, "minuend": 42}, "id": 3}
```

to figure out which format the request uses, we first [unmarshal](https://github.com/NethermindEth/juno/pull/763/files#diff-b18e347b821e9f1511d44c3ccb83f52bfe315b2a648678802c06781bd62b8e9fL306-L321) the entire "params" field in to a `any` typed variable. That variable either ends up being a `[]any` or `map[string]any`

then each element of the list or map needs to be converted to the type that the handler expects (`t` in this piece of code). So we need a way to convert an object in type `any` to a some arbitrary concrete type `t`. Only way that I could think of when I first wrote this JSONRPC server was to marshal the `any` value back to JSON and then unmarshal it to type `t`

Reply:
This is out of scope for this PR, but feel free to suggest any alternatives. It would be really nice to get rid of this round trip to and from JSON.